### PR TITLE
Add new JDBC 4.1 methods (java 1.7)

### PR DESCRIPTION
**Pull request by rpastrana**

### CODE REVIEW

**Comment by rpastrana** [comment]
@garonsky  please review.

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	<groupId>org.hpccsystems</groupId>
 	<artifactId>jdbcdriver</artifactId>
 	<!-- When updating the version number please make the corresponding changes in version.cmake -->
-	<version>0.2.6-SNAPSHOT</version>
+	<version>0.3.0-SNAPSHOT</version>
```
Comment (garonsky) [inline pom.xml]: Should this be in its own pull request?

File: version.cmake [hunk 2]
```diff
@@ -4,8 +4,8 @@
 ## NOTE: when updating the version info please make the corresponding changes in pom.xml 
 ###
 set ( HPCC_MAJOR 0 )
-set ( HPCC_MINOR 2 )
-set ( HPCC_POINT 6 )
+set ( HPCC_MINOR 3 )
+set ( HPCC_POINT 0 )
```
Comment (garonsky) [inline version.cmake]: Is the update to the version of cmake required for this committ?

File: src/main/java/org/hpccsystems/jdbcdriver/HPCCResultSet.java [hunk 3]
```diff
@@ -2246,4 +2246,16 @@ else if (docElement.getElementsByTagName("Exception").getLength() > 0)
 
         return rowcount;
     }
+
+    //Introduced in java 1.7 @Override
+    public <T> T getObject(int columnIndex, Class<T> type) throws SQLException
+    {
+        return null;
+    }
```
Comment (garonsky) [inline src/main/java/org/hpccsystems/jdbcdriver/HPCCResultSet.java]: Should this be implemented?

File: src/main/java/org/hpccsystems/jdbcdriver/HPCCResultSet.java [hunk 4]
```diff
@@ -2246,4 +2246,16 @@ else if (docElement.getElementsByTagName("Exception").getLength() > 0)
...
+    public <T> T getObject(int columnIndex, Class<T> type) throws SQLException
+    {
+        return null;
+    }
+
+    //Introduced in java 1.7 @Override
+    public <T> T getObject(String columnLabel, Class<T> type) throws SQLException
+    {
+        return null;
+    }
```
Comment (garonsky) [inline src/main/java/org/hpccsystems/jdbcdriver/HPCCResultSet.java]: Should this be implemented?

**Comment by garonsky** [comment]
The rest looks good.

File: pom.xml (hunk 1 above)
Comment (rpastrana) [inline pom.xml]: Yes, will remove from this commit

File: version.cmake (hunk 2 above)
Comment (rpastrana) [inline version.cmake]: It's showing up in this commit, but this change was actually done in a different commit. Will rebase branch
